dotnet/samples/Demos/ProcessFrameworkWithAspire/ProcessFramework.Aspire/ProcessFramework.Aspire.SummaryAgent/Program.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Azure.AI.OpenAI;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Agents;
+using Microsoft.SemanticKernel.ChatCompletion;
+using OpenTelemetry;
+using OpenTelemetry.Exporter;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+using ProcessFramework.Aspire.Shared;
+
+var builder = WebApplication.CreateBuilder(args);
+
+string otelExporterEndpoint = builder.GetConfiguration("OTEL_EXPORTER_OTLP_ENDPOINT");
+string otelExporterHeaders = builder.GetConfiguration("OTEL_EXPORTER_OTLP_HEADERS");
+
+AppContext.SetSwitch("Microsoft.SemanticKernel.Experimental.GenAI.EnableOTelDiagnosticsSensitive", true);
+
+var loggerFactory = LoggerFactory.Create(builder =>
+{
+    // Add OpenTelemetry as a logging provider
+    builder.AddOpenTelemetry(options =>
+    {
+        options.AddOtlpExporter(exporter => { exporter.Endpoint = new Uri(otelExporterEndpoint); exporter.Headers = otelExporterHeaders; exporter.Protocol = OtlpExportProtocol.Grpc; });
+        // Format log messages. This defaults to false.
+        options.IncludeFormattedMessage = true;
+    });
+
+    builder.AddTraceSource("Microsoft.SemanticKernel");
+    builder.SetMinimumLevel(LogLevel.Information);
+});
+
+using var traceProvider = Sdk.CreateTracerProviderBuilder()
+    .AddSource("Microsoft.SemanticKernel*")
+    .AddOtlpExporter(exporter => { exporter.Endpoint = new Uri(otelExporterEndpoint); exporter.Headers = otelExporterHeaders; exporter.Protocol = OtlpExportProtocol.Grpc; })
+    .Build();
+
+using var meterProvider = Sdk.CreateMeterProviderBuilder()
+    .AddMeter("Microsoft.SemanticKernel*")
+    .AddOtlpExporter(exporter => { exporter.Endpoint = new Uri(otelExporterEndpoint); exporter.Headers = otelExporterHeaders; exporter.Protocol = OtlpExportProtocol.Grpc; })
+    .Build();
+
+builder.AddServiceDefaults();
+builder.AddAzureOpenAIClient("openAiConnectionName");
+builder.Services.AddSingleton(builder =>
+{
+    var kernelBuilder = Kernel.CreateBuilder();
+
+    kernelBuilder.AddAzureOpenAIChatCompletion("gpt-4o", builder.GetService<AzureOpenAIClient>());
+
+    return kernelBuilder.Build();
+});
+
+var app = builder.Build();
+
+app.UseHttpsRedirection();
+
+app.MapPost("/api/summaryagent", async (Kernel kernel, SummarizeRequest summarizeRequest) =>
```

Comment:
nit: it's a bit confusing that the API path contains the same `summaryagent` string used also for the service name. Perhaps it could be a verb like `summarize`?  similarly for `/api/translatoragent`

Reply:
I wanted to make it clear that the API uses an agent under the hood.